core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java

```diff
@@ -3948,6 +3950,30 @@ private void convertSelectList(
     }
   }
 
+  /**
+   * Convert the table function in select to Join or TableFunctionScan.
+   * @param bb       Blackboard
+   * @param select   SqlSelect
+   * */
+  private void convertTableFunctionInSelect(Blackboard bb, SqlSelect select) {
+    SqlBasicCall tableFunction = validator.getTableFunctionInSelect(select);
+    // rewrite the table function if select list contain one
+    if (tableFunction != null) {
```

Comment:
Under what conditions it will become a Join?

Reply:
If select cotains a table function, It will be converted to a join.